esp-config/src/bin/esp-config/main.rs

```diff
@@ -0,0 +1,332 @@
+    }
+
+    let mut configs = Vec::new();
+    let chip = esp_metadata::Config::for_chip(chip.as_ref().unwrap());
+    let features = vec![];
+    for krate in meta.packages {
+        let maybe_cfg = krate.manifest_path.parent().unwrap().join("esp_config.yml");
+        if maybe_cfg.exists() {
+            let yaml = std::fs::read_to_string(&maybe_cfg)?;
+            let (cfg, options) =
+                esp_config::evaluate_yaml_config(&yaml, Some(chip.clone()), features.clone(), true)
+                    .map_err(|e| {
+                        format!(
+                            "Error evaluating YAML config for crate {}: {}",
+                            krate.name, e
+                        )
+                    })?;
+
+            let crate_name = cfg.krate.clone();
+
+            let options: Vec<ConfigItem> = options
+                .iter()
+                .map(|cfg| {
+                    Ok::<ConfigItem, Box<dyn Error>>(ConfigItem {
+                        option: cfg.clone(),
+                        actual_value: {
+                            let key = format!(
+                                "{}_CONFIG_{}",
+                                crate_name.clone().to_ascii_uppercase().replace("-", "_"),
+                                cfg.name.to_ascii_uppercase().replace("-", "_")
+                            );
+                            let def_val = &cfg.default_value.to_string();
+                            let val = envs.get(&key).unwrap_or(def_val);
+
+                            parse_value_from_string(val, &cfg.default_value) //
+                                .map_err(|_| {
+                                    <std::string::String as Into<Box<dyn Error>>>::into(format!(
+                                        "Unable to parse '{val}' for option '{}'",
+                                        &cfg.name
+                                    ))
+                                })?
+                        },
+                    })
+                })
+                .collect::<Result<Vec<ConfigItem>, Box<dyn Error>>>()?;
+
+            configs.push(CrateConfig {
+                name: crate_name.clone(),
+                checks: cfg.checks.clone(),
+                options,
+            });
+        }
+    }
+
+    if configs.is_empty() {
+        return Err("No config files found.".into());
+    }
+
+    Ok(configs)
+}
+
+fn check_after_changes(
+    work_dir: &PathBuf,
+    chip_from_args: Option<esp_metadata::Chip>,
+    config_file: Option<String>,
+) -> Result<Option<String>, Box<dyn Error>> {
+    println!("Check configuration...");
+
+    let configs = parse_configs(work_dir, chip_from_args, config_file)?;
+
+    for config in configs {
+        if let Some(value) = tui::validate_config(&config) {
+            return Err(value.into());
+        }
+    }
+
+    Ok(None)
+}
+
+fn parse_i128(str: &str) -> Result<i128, std::num::ParseIntError> {
```

Comment:
This is different from the library-side parsing logic (it doesn't support octal format). Let's try to reuse `Value::parse_in_place`.